### DISCUSSION TITLE
Edit fields (title, author, etc.)

### DISCUSSION THREAD

**Discussion by Shohreh** [sumatrapdfreader/sumatrapdf#3933] [Ideas]
Hello,

I sometimes need to edit a PDF's fields such as author or title.

Currently, I use cpdf, but it'd be easier if SumatraPDF displayed a form instead.

```
c:\cpdf\cpdf.exe -set-title "blah" "input.pdf" -o "temp1.pdf"
c:\cpdf\cpdf.exe -set-subject "blah" "temp1.pdf" -o "temp2.pdf"
c:\Apps\cpdf\cpdf.exe -set-author "blah" "temp2.pdf" -o "temp3.pdf"
del input.pdf
ren temp3.pdf input.pdf
del temp*.pdf
```

Cheers,

**Comment by GitHubRulesOK** [comment]
Not possible to edit a PDF metadata without rewrite the file (edit could need multiple locations as its not always held in just one place) annotation is different as that appending to source file data. In theory SumatraPDF could append new data but that would not remove  old author. Any way if the Author wrote the PDF correctly with their "Title" those should be left as they are.

**Reply by Shohreh** [reply]
Too bad. Thanks for the infos.

**Reply by GitHubRulesOK** [reply]
Note that exifTool clearly state when you edit pdf metadata then the old data is retained, https://exiftool.org/TagNames/PDF.html
>All metadata edits are reversible. While this would normally be considered an advantage, it is a potential security problem because old information is never actually deleted from the file.   

To edit it directly you need to decompress and use find and replace. see qpdf qdf mode (but potentially that can break a file).
If cpdf does it correctly (they should) I would stick with that.

**Reply by Shohreh** [reply]
Thanks. I'll write a small GUI to make it easier to edit fields.

**Comment by allanmaxd** [comment]
Want to edit your PDF files online???

**Reply by GitHubRulesOK** [reply]
Allan your editor seems not to handle more than Edge does, that much I can do in a local copy, so am I missing some meta-editor feature ? SumatraPDF editor has annotation and can edit the nested PDF too just not using its online version.

![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/7824d4cc-803f-48d2-9452-1dc8e2e02870)
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/61884bb7-b40b-4e42-bfa5-d96ac10462e6)
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/193d346f-7a9c-435a-9415-60eb6e7caaf8)

**Comment by kjk** [comment]
Should be a feature request in issues. Bug reports / feature requests in discussions will be forgotten.
